updates-and-signals/introduction/src/workflows.ts

```diff
@@ -0,0 +1,126 @@
+import * as wf from '@temporalio/workflow';
+import { Mutex } from 'async-mutex';
+
+export enum Language {
+  ARABIC = 'ARABIC',
+  CHINESE = 'CHINESE',
+  ENGLISH = 'ENGLISH',
+  FRENCH = 'FRENCH',
+  HINDI = 'HINDI',
+  PORTUGUESE = 'PORTUGUESE',
+  SPANISH = 'SPANISH',
+}
+
+interface GetLanguagesInput {
+  includeUnsupported: boolean;
+}
+
+interface ApproveInput {
+  name: string;
+}
+
+// 👉 Use the objects returned by defineQuery/defineSignal/defineUpdate to set
+// the message handlers in Workflow code, and to send messages from Client code.
+export const getLanguages = wf.defineQuery<Language[], [GetLanguagesInput]>('getLanguages');
+export const approve = wf.defineSignal<[ApproveInput]>('approve');
+export const setLanguage = wf.defineUpdate<Language, [Language]>('setLanguage');
+export const setLanguageUsingActivity = wf.defineUpdate<Language, [Language]>('setLanguageUsingActivity');
+export const getLanguage = wf.defineQuery<Language>('getLanguage');
+
+export async function greetingWorkflow(): Promise<string> {
+  // The workflow has built-in support for two languages only (see remote
+  // greeting service below for support for more languages).
+  const greetings: Partial<Record<Language, string>> = {
```

Comment:
@mjameswh do you see a better way to type this? The idea is that this object holds translations that are available in-workflow, and that the activity can be used to obtain other translations. I hesitated to create the object with most values equal to empty-string, since the code is used as a docs snippet and it looked a bit distracting.